File: api/garmin-sync.py
import json
import os
from http.server import BaseHTTPRequestHandler
from garminconnect import Garmin
from datetime import date

class handler(BaseHTTPRequestHandler):
    def do_POST(self):
        length = int(self.headers.get('Content-Length', 0))
        body = json.loads(self.rfile.read(length))
        
        email = body.get('email')
        password = body.get('password')
        
        if not email or not password:
            self.send_response(400)
            self.end_headers()
            self.wfile.write(json.dumps({'error': 'Missing credentials'}).encode())
            return
        
        try:
            client = Garmin(email, password)
            client.login()
            
            today = date.today().isoformat()
            stats = client.get_stats(today)
            hrv = client.get_hrv_data(today)
            
            result = {
                'hrv': hrv.get('hrvSummary', {}).get('lastNight'),
                'body_battery': stats.get('bodyBatteryMostRecentValue'),
                'sleep_score': stats.get('sleepingSeconds'),
                'resting_hr': stats.get('restingHeartRate'),
                'stress': stats.get('averageStressLevel'),
            }
            
            # Get training readiness
            try:
                readiness = client.get_training_readiness(today)
                if readiness:
                    result['training_readiness'] = readiness[0].get('score')
            except:
                pass

            self.send_response(200)
            self.send_header('Content-Type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps(result).encode())
            
        except Exception as e:
            self.send_response(401)
            self.end_headers()
            self.wfile.write(json.dumps({'error': str(e)}).encode())

File: api/garmin-sync.js
import { GarminConnect } from 'garmin-connect'

export default async function handler(req, res) {
  if (req.method !== 'POST') return res.status(405).end()
  
  const { email, password } = req.body
  if (!email || !password) return res.status(400).json({ error: 'Missing credentials' })

  try {
    const client = new GarminConnect({ username: email, password })
    await client.login()
    
    const today = new Date()
    const dateStr = today.toISOString().split('T')[0]

    const [sleep, hr, steps] = await Promise.allSettled([
      client.getSleepData(today),
      client.getHeartRate(today),
      client.getSteps(today),
    ])

    // Try wellness and HRV via direct get with correct base
    const [wellness, hrv, readiness] = await Promise.allSettled([
      client.get(`https://connect.garmin.com/wellness-service/wellness/dailySummary/${dateStr}`),
      client.get(`https://connect.garmin.com/hrv-service/hrv/${dateStr}`),
      client.get(`https://connect.garmin.com/training-readiness-service/training-readiness/${dateStr}`),
    ])

    res.json({
      debug: {
        sleep: sleep.status === 'fulfilled' ? sleep.value : sleep.reason?.message,
        hr: hr.status === 'fulfilled' ? hr.value : hr.reason?.message,
        wellness: wellness.status === 'fulfilled' ? wellness.value : wellness.reason?.message,
        hrv: hrv.status === 'fulfilled' ? hrv.value : hrv.reason?.message,
        readiness: readiness.status === 'fulfilled' ? readiness.value : readiness.reason?.message,
      }
    })
  } catch(e) {
    res.status(401).json({ error: e.message })
  }
}
